src/Symfony/Bridge/Twig/Command/LintCommand.php

```diff
@@ -77,15 +77,16 @@ protected function execute(InputInterface $input, OutputInterface $output)
     {
         $io = new SymfonyStyle($input, $output);
         $filenames = $input->getArgument('filename');
-        $hasStdin = ['-'] === $filenames;
 
-        if ($hasStdin || !$filenames) {
-            if ($hasStdin || 0 === ftell(STDIN)) { // remove 0 === ftell(STDIN) check in 5.0
-                if (!$hasStdin) {
-                    @trigger_error('Calling to the "lint:twig" command providing pipe file content to STDIN without passing the dash symbol "-" explicitly is deprecated since Symfony 4.4.', E_USER_DEPRECATED);
+        if (!$filenames || ['-'] === $filenames) {
+            if (null !== $stdin = $this->getStdin()) {
+                if (!$filenames) {
+                    @trigger_error('Piping content from STDIN to the "lint:twig" command without passing the dash symbol "-" as argument is deprecated since Symfony 4.4.', E_USER_DEPRECATED);
                 }
 
-                return $this->display($input, $output, $io, [$this->validate($this->getStdin(), uniqid('sf_', true))]);
+                return $this->display($input, $output, $io, [$this->validate($stdin, uniqid('sf_', true))]);
+            } elseif (['-'] === $filenames) {
```

Comment:
I don't think this is correct: when `-` is explicit, we must not call ftell and must block on stdin until EOF

Reply:
removed

Reply:
you removed the exception. But you did not implement the fixed behavior, as you still use ftell for that case.